src/main/java/com/bmuschko/gradle/docker/tasks/image/DockerRemoveImage.java

```diff
@@ -16,29 +16,151 @@
 package com.bmuschko.gradle.docker.tasks.image;
 
 import com.github.dockerjava.api.command.RemoveImageCmd;
+import org.gradle.api.provider.ListProperty;
 import org.gradle.api.provider.Property;
+import org.gradle.api.provider.Provider;
 import org.gradle.api.tasks.Input;
 import org.gradle.api.tasks.Optional;
 
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.Callable;
+
 public class DockerRemoveImage extends DockerExistingImage {
 
+    /**
+     * Configures `--force` option when removing the images.
+     */
     @Input
     @Optional
     public final Property<Boolean> getForce() {
         return force;
     }
 
+    /**
+     * Configures `--no-prune` option when removing the images.
+     */
+    @Input
+    @Optional
+    public final Property<Boolean> getNoPrune() {
+        return noPrune;
+    }
+
     private final Property<Boolean> force = getProject().getObjects().property(Boolean.class);
+    private final Property<Boolean> noPrune = getProject().getObjects().property(Boolean.class);
+
+    private final ListProperty<String> images = getProject().getObjects().listProperty(String.class);
+
+    /**
+     * Sets the IDs or names of the images to be removed.
+     *
+     * @param images the names or IDs of the images.
+     * @see #images(ListProperty)
+     */
+    public void images(List<String> images) {
+        this.images.set(images);
+    }
+
+    /**
+     * Sets the IDs or names of the images to be removed.
+     *
+     * @param images the names or IDs of the images.
+     * @see #images(ListProperty)
+     */
+    public void images(String... images) {
+        this.images.set(Arrays.asList(images));
+    }
+
+    /**
+     * Sets the IDs or names of the images to be removed.
+     *
+     * @param images Image ID or name as {@link Callable}
+     * @see #images(String...)
+     * @see #images(ListProperty)
+     */
+    public void images(Callable<List<String>> images) {
+        // TODO: How to do this, and is it necessary?
+        // images(providers.provider(images));
+    }
+
+    /**
+     * Sets the IDs or names of the images to be removed.
+     *
+     * @param images Image ID or name as {@link Provider}
+     * @see #images(String...)
+     */
+    public void images(ListProperty<String> images) {
+        this.images.set(images);
+    }
+
+    // Overriding methods to provide a warning message.
```

Comment:
Remove all of those methods and extend from `AbstractDockerRemoteApiTask` instead. The parent class `DockerExistingImage` is really only meant for a single image reference.